pootle/apps/pootle_project/views.py

```diff
@@ -64,21 +64,24 @@ def get_object(self):
         if not (self.kwargs["dir_path"] or self.kwargs["filename"]):
             return self.project
 
-        regex = (
-            r"^/.*/%s/%s"
+        project_path = (
+            "/%s/%s%s"
             % (self.project.code,
-               self.kwargs['dir_path']))
-
+               self.kwargs['dir_path'],
+               self.kwargs['filename']))
+        regex = r"^/[^/]*%s$" % project_path
         if not self.kwargs["filename"]:
             dirs = Directory.objects.live().select_related("parent__pootle_path")
             resources = (
-                dirs.filter(pootle_path__regex="%s$" % regex)
-                    .exclude(pootle_path__startswith="/templates"))
+                dirs.exclude(pootle_path__startswith="/templates")
+                    .filter(pootle_path__endswith=project_path)
+                    .filter(pootle_path__regex=regex))
```

Comment:
I read https://thingsilearn.wordpress.com/2008/02/28/mysql-query-speed-regexp-vs-like/ but sadly it seems that we can't get rid of the regex altogether.